### PR TITLE
Unique logins for users

### LINKED ISSUE #698: Unique logins for users
Move login to indexable fields and allow getting user via login in URLs and APIs

* Should be optional currently but strictly unique
* Should be present when creating users in UI

### PR DESCRIPTION
**Pull request by DenisVorop**
## PR includes

-   [x] Feature

## Related issues

Resolve #698 

## QA Instructions, Screenshots, Recordings

Снимок экрана 2024-04-25 в 14 01 38

### CODE REVIEW

File: src/modules/userMethods.ts [hunk 1]
```diff
@@ -185,6 +186,17 @@ export const userMethods = {
...
 
+    getByLogin: async (
+        login: string,
+        sessionUser: SessionUser,
+    ): Promise<User & UserMemberships & UserSupervisor & UserAchievements & UserSupervisorOf & UserSupervisorIn> => {
+        const user = await prisma.user.findUnique({ where: { login } });
+
+        if (!user) throw new TRPCError({ code: 'NOT_FOUND', message: `No user with login ${login}` });
+
+        return userMethods.getById(user.id, sessionUser);
```
Comment (9teen90nine) [inline src/modules/userMethods.ts:197]: why would you query second time? you already got the user

Comment (DenisVorop) [reply src/modules/userMethods.ts:197]: i get only id, and use request `getById` to get the same user with needed includes and schema

**Review by 9teen90nine — APPROVED** [review]

File: src/modules/userMethods.ts [hunk 2]
```diff
@@ -134,6 +134,7 @@ export const userMethods = {
                 name: trimAndJoin([data.surname, data.firstName, data.middleName]),
                 email: data.email,
                 supervisorId: data.supervisorId,
+                login: data.login,
```
Comment (tsumo) [inline src/modules/userMethods.ts:137]: login service (line 115) is not needed anymore

**Review by tsumo — CHANGES_REQUESTED** [review]

File: src/modules/userMethods.ts (hunk 2 above)
Comment (tsumo) [reply src/modules/userMethods.ts:137]: It's also in seed

**Review by tsumo — APPROVED** [review]

**Pull request merged**